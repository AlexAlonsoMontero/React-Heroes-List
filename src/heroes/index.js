//EXPORT PAGES
export * from './pages/MarvelPages';
export * from './pages/DcPages';
export * from './pages/SearchPages';
export * from './pages/HeroPage.jsx'


//ExPORT  ROUTES
export * from './routes/HeroesRoutes';
