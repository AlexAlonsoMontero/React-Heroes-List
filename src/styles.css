body {
    background-image: url('../assets/generic-fondo.jpg');
    background-repeat: no-repeat;
    background-size:cover;
    padding-bottom: 50px;
    min-height: 100vh;
}

h1,h3, h4 {
    color: white;
    margin-top: 5px;
}



hr{
    color: white;
}