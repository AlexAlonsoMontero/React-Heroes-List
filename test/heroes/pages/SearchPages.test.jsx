import { fireEvent, getAllByRole, getByRole, render, screen } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import { SearchPages } from "../../../src/heroes";
const mockNavigate = jest.fn();

jest.mock('react-router-dom', ()=>({
    ...jest.requireActual('react-router-dom'),
    useNavigate: ()=>mockNavigate
}));
describe('Pruebas en <SearchPage />',()=>{
    test('debe de mostrarse corretamtne con valores por defecto', ()=>{
        const { container } = render(
            <MemoryRouter>
                <SearchPages />
            </MemoryRouter>
        )
        
        expect( container ).toMatchSnapshot();
    })
    test('debe de mostrarsa Batman y el input con el valor del query string', ()=>{
        render(
            <MemoryRouter initialEntries={['/search?q=batman']}>
                <SearchPages />
            </MemoryRouter>
        )
        
        const input = screen.getByRole('textbox');
        expect ( input.value.toString() ).toBe('batman');
        const img =  screen.getByRole('img');
        expect( img.src ).toContain('batman');
        const div = screen.getByLabelText('container');

    })

    test('debe de mostrar un error si no se encuentra el heroe',()=>{
        render(
            <MemoryRouter initialEntries={['/search?q=asdfafdaf']}>
                <SearchPages />
            </MemoryRouter>
        )
        expect( screen.getByText('There is not a hero')).toBeTruthy();


    })

    test('debe de llamar a la pantalla nueva el navigate', ()=>{
        const inputValue = 'superman'
        render (
            <MemoryRouter initialEntries={['/search']}>
                <SearchPages />
            </MemoryRouter>
        )
        const input = screen.getByRole('textbox');
        fireEvent.change(input,{target:{name: 'searchText', value: inputValue}});
        const form = screen.getByLabelText("form");
        fireEvent.submit(form)
        expect(mockNavigate).toHaveBeenCalledWith(`?q=${inputValue}`)

    })
})