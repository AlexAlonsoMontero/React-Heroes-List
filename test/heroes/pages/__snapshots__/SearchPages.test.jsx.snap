// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <SearchPage /> debe de mostrarse corretamtne con valores por defecto 1`] = `
<div>
  <h1>
    SearchPages
  </h1>
  <hr />
  <div
    class="row"
  >
    <div
      class="col-5"
    >
      <h4>
        Searching
      </h4>
      <hr />
      <form
        aria-label="form"
      >
        <input
          autocomplete="off"
          class="form-control"
          name="searchText"
          placeholder="Search a hero"
          type="text"
          value=""
        />
        <button
          class="btn btn-dark mt-2"
        >
          Search
        </button>
      </form>
    </div>
    <div
      class="col-7"
    >
      <h4>
        Results
      </h4>
      <hr />
      <div
        class="alert alert-dark"
      >
        Search a hero
      </div>
      <div
        aria-label="container"
        class="alert alert-danger"
        style="display: none;"
      >
        There is not a hero 
        <b>
           
           
        </b>
      </div>
      <div />
    </div>
  </div>
</div>
`;
